src/app/content/highlights/components/CardWrapper.tsx

```diff
@@ -11,20 +15,109 @@ interface Props {
   className?: string;
 }
 
+const displayAllCardsMediaQuery = `(max-width: ${mediaQueryBreakToStopDisplaingAllCards})`;
+
 // tslint:disable-next-line:variable-name
-const Wrapper = ({highlights, className, container, highlighter}: Props) => highlights.length
-  ? <div className={className}>
+const Wrapper = ({highlights, className, container, highlighter}: Props) => {
+  const displayAllCards = !assertWindow().matchMedia(displayAllCardsMediaQuery).matches;
+
+  const element = React.useRef<HTMLElement>(null);
+  const [cardsPositions, setCardsPositions] = React.useState<Map<string, number>>(new Map());
+  const [cardsHeights, setCardsHeights] = React.useState<Map<string, number>>(new Map());
+
+  const onHeightChange = (id: string, ref: React.RefObject<HTMLElement>) => {
+    if (!displayAllCards) { return; }
+
+    const height = ref.current && ref.current.offsetHeight;
+    if (cardsHeights.get(id) !== height) {
+      setCardsHeights((data) => new Map(data.set(id, height === null ? 0 : height)));
+    }
+  };
+
+  const onFocus = (id: string) => {
+    if (!displayAllCards) { return; }
+
+    const highlight = highlights.find((search) => search.id === id);
+    const position = cardsPositions.get(id);
+    if (typeof position !== 'number' || !highlight) { return; }
+
+    const topOffset = assertDefined(
+      getHighlightTopOffset(container, highlight),
+      `Couldn't get top offset for highlights`
+    );
+
+    if (position > topOffset) {
+      element.current!.style.top = `-${position - topOffset}px`;
```

Comment:
I thought this would be easier to implement for now and in the next step we could just modify how `updatePositions` behave when we pass additional parameter like `focusId: highlightId`

Reply:
in that slack thread others are ok with this current behavior, so lets leave it the way you have it